content/blog/2021/06/kwctl-intro-for-kubernetes-administrators.md

```diff
@@ -0,0 +1,307 @@
+simple command:
+
+```shell
+$ kwctl pull registry://ghcr.io/kubewarden/policies/psp-capabilities:v0.1.4
+```
+
+This downloads the policy and stores it under Alice's home directory.
+
+## Understanding how the policy works
+
+Alice now needs to understand how to use the policy. Each policy listed on
+Kubewarden's Policy Hub has links pointing to their documentation.
+
+Instead of opening one of these links, Alice prefers to consult the
+metadata provided by the policy:
+
+```shell
+$ kwctl inspect registry://ghcr.io/kubewarden/policies/psp-capabilities:v0.1.4
+``` 
+
+This command produces the following output:
+
+![Output of kwctl inspect](/images/2021-june-kwctl-intro-admin-inspect-output.png)
+
+> **Note well:** a Kubewarden policy is a regular "`.wasm`" file that, in addition
+> to contain the policy's bytecode, it also embeds additional metadata.
+>
+> This allows to have everything in a **single place**, **versioned together** and
+> **available off-line**.
+
+## Evaluating the policy
+
+Now Alice wants to assess the reliability of the policy, plus find the right
+configuration values to satisfy her requirements.
+
+There are some good news for Alice: she doesn't need to deploy the policy into
+a Kubernetes cluster to perform this kind of rapid iterations. Once again, kwctl
+can help her with the `run` subcommand.
+
+The `kwctl run` command takes the following parameters:
+  * **`-r <request-path>`**: path to a file containing the Kubernetes Admission Request
+    to be evaluated
+  * **`--settings-json "<JSON document>"`**: a string containing the JSON
+    representation of policy's settings. It's also possible to use the
+    "**`-s <settings-path>`**" flag to read the policy settings from a local file
+  * **`<URI of the policy>`**: the URI pointing to the policy to be used
+
+> **Note well:** unfortunately, obtaining Kubernetes Admission Requests
+> requires some extra work. This is a "Kubernetes problem", we plan to
+> address it in the near future.
+>
+> In the meantime, we will assume Alice has access to files like
+> [these ones](https://github.com/kubewarden/psp-capabilities/tree/a786aba746e807aa3c6121438e8c05e724400861/test_data).
+
+Alice will use [this request](https://raw.githubusercontent.com/kubewarden/psp-capabilities/a786aba746e807aa3c6121438e8c05e724400861/test_data/req_pod_with_container_with_capabilities_added.json);
+that attempts to create a Pod with a container that has the
+following `securityContext` configuration:
+
+  * Add the `NET_ADMIN` and the `SYS_TIME` capabilities
+  * Drop the `SYS_PTRACE` capability
+
+As a first step, Alice defines a policy that allows containers to add only the `SYS_TIME` capability:
+
+```shell
+$ kwctl run registry://ghcr.io/kubewarden/policies/psp-capabilities:v0.1.4 \
+            -r req_pod_with_container_with_capabilities_added.json \
+            --settings-json '{"allowed_capabilities": ["SYS_TIME"]}'
+```
+
+The output of the command, piped into `jq`, is the following one:
```

Comment:
What is `jq` and where can I get it ?

Reply:
This is a well known command, I've made `jq` a link to its homepage

Reply:
Thanks for the link, jq wasn't known to me.